core/src/main/scala/uk/gov/nationalarchives/tdr/api/core/db/dao/FileFormatDao.scala

```diff
@@ -14,9 +14,23 @@ class FileFormatDao(implicit val executionContext: ExecutionContext) {
 
   private val insertQuery = fileFormats returning fileFormats.map(_.id) into ((fileFormat, id) => fileFormat.copy(id = Some(id)))
 
-  def create(pronomId: String, fileId: Int): Future[FileFormat] = {
-    val fileFormat: FileFormat = FileFormat(null, pronomId, fileId)
-    db.run(insertQuery += fileFormat)
+
+  def getByFileId(fileId: Int): Future[Option[FileFormat]] = {
+    db.run(fileFormats.filter(_.fileId === fileId).result).map(_.headOption)
+  }
+
+  def createOrUpdate(pronomId: String, fileId: Int) = {
```

Comment:
I wonder if it would be better to have separate processes for creating and updating file statuses?

If we create the file status row in the FileDao at the same time as the file, we can put them in the same transaction so they both succeed or fail together.

Reply:
We don't need to create them at the same time though. We need the file status to store the client checksum but this table isn't needed until later in the process and a missing row in this table would be a good indicator that something has gone wrong in the process.

Reply:
I suspect it'll be less error-prone to just create the row when we create the file, but for the prototype it doesn't really matter. :slightly_smiling_face: